### PR TITLE
frontend/feat: Making auto rickshaw image configurable

### PR DESCRIPTION
**Pull request by MercyQueen**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### CODE REVIEW

File: Frontend/ui-customer/src/Services/Backend.purs [hunk 1]
```diff
@@ -704,19 +704,25 @@ type Markers = {
...
     destMarker : if isPreviousVersion (getValueToLocalStore VERSION_NAME) (getPreviousVersion "") then "src_marker" else "ny_ic_src_marker"
 }
 
-rideTracking :: String -> Markers
-rideTracking variant = {
-    srcMarker : if isPreviousVersion (getValueToLocalStore VERSION_NAME) (getPreviousVersion "") then "ny_ic_auto_map" 
+rideTracking :: String -> Maybe String-> Markers
+rideTracking variant city = {
+    srcMarker : case city of 
+        Just "std:040" -> "ny_ic_black_yellow_auto" -- for hyderabad
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Services/Backend.purs:721]: we should refactor this

File: Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java [hunk 2]
```diff
@@ -1275,6 +1275,7 @@ public void removeAllPolylines(String str) {
         ExecutorManager.runOnMainThread(() -> {
             removeMarker("ic_auto_nav_on_map");
             removeMarker("ny_ic_vehicle_nav_on_map");
+            removeMarker("ny_ic_black_yellow_auto");
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java:1278]: can we send a array of marker names needs to be removed as param to this? add all these as falback and mention deprecated

File: Frontend/ui-common/src/Resource/Constants.purs [hunk 3]
```diff
@@ -1,9 +1,140 @@
...
 import Prelude ((==))
+import Common.Types.App(CityConfig)
 
 zoomLevel :: Number
 zoomLevel = if (os == "IOS") then 19.0 else 17.0
 
 pickupZoomLevel :: Number
 pickupZoomLevel = 17.0
+
+cityConfig :: Array CityConfig
```
Comment (Vignesh-772) [inline Frontend/ui-common/src/Resource/Constants.purs:12]: Can we have a new Module for CityConfigs ?

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 4]
```diff
@@ -1212,11 +1212,14 @@ configurations =
...
               }
 
-getVehicleImage :: String -> String -> String
-getVehicleImage variant vehicleDetail = do
+getVehicleImage :: String -> String -> Maybe String -> String
+getVehicleImage variant vehicleDetail city = do
   let details = (toLower vehicleDetail)
   fetchImage FF_ASSET $ 
-    if (variant == "AUTO_RICKSHAW") then "ic_auto_rickshaw"
+    if (variant == "AUTO_RICKSHAW") then do
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:1219]: remove extra brackets

File: Frontend/ui-customer/src/Flow.purs [hunk 5]
```diff
@@ -294,6 +294,9 @@ currentRideFlow rideAssigned = do
         _ <- pure $ spy "Active api" listResp
         modifyScreenState $ HomeScreenStateType (\homeScreen → newState)
         updateLocalStage rideStatus
+        case (getFlowStatusData "LazyCheck") of
+            Just (FlowStatusData flowStatusData) -> modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{city = flowStatusData.source.city}})
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:298]: use `maybe` function here also

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**